photutils/psf/psfphotometry.py

```diff
@@ -0,0 +1,282 @@
+        """
+
+        result_tab = Table([[], [], [], [], []],
+                           names=('id', 'group_id', 'x_fit', 'y_fit',
+                                  'flux_fit'),
+                           dtype=('i4', 'i4', 'f8', 'f8', 'f8'))
+
+        star_groups = star_groups.group_by('group_id')
+        
+        groups_order = []
+        for g in star_groups.groups:
+            groups_order.append(len(g))
+
+        N = len(groups_order)
+        while N > 0:
+            curr_order = np.min(groups_order)
+            n = 0
+            N = len(groups_order)
+            while(n < N):
+                if curr_order == len(star_groups.groups[n]):
+                    group_psf = self._get_sum_psf_model(star_groups.groups[n])
+                    x, y, data = self._get_shape_and_data(shape=self.fitshape,\
+                                             star_group=star_groups.groups[n],
+                                                          image=image)
+                    fit_model = self.fitter(group_psf, x, y, data)
+                    print(fit_model)
+                    param_table = self._model_params2table(fit_model,\
+                                                        star_groups.groups[n])
+                    result_tab = vstack([result_tab, param_table])
+                    image = subtract_psf(image, self.psf, param_table)
+                    N = N - 1
+                n += 1
+        return result_tab, image
+
+    def _model_params2table(self, fit_model, star_group):
+        """
+        Place fitted parameters into an astropy table.
+        
+        Parameters
+        ----------
+        fit_model : Fittable2DModel
+        star_group : ~astropy.table.Table
+        
+        Returns
+        -------
+        param_tab : ~astropy.table.Table
+            Table that contains the fitted parameters.
+        """
+
+        param_tab = Table([[], [], [], [], []],
+                          names=('id', 'group_id', 'x_fit', 'y_fit',
+                                 'flux_fit'),
+                          dtype=('i4', 'i4', 'f8', 'f8', 'f8'))
+        if np.size(fit_model) == 1:
+            param_tab.add_row([[star_group['id'][0]],
+                               [star_group['group_id'][0]],
+                               [getattr(fit_model,'x_0').value],
+                               [getattr(fit_model, 'y_0').value],
+                               [getattr(fit_model, 'flux').value]])
+        else:
+            for i in range(np.size(fit_model)):
+                param_tab.add_row([[star_group['id'][i]],
+                                   [star_group['group_id'][i]],
+                                   [getattr(fit_model,'x_0_'+str(i)).value],
+                                   [getattr(fit_model, 'y_0_'+str(i)).value],
+                                   [getattr(fit_model, 'flux_'+str(i)).value]])
+        return param_tab
+
+    def _get_shape_and_data(self, shape, star_group, image):
+        """
```

Comment:
I'm not sure is `extract_array` itself is the the thing to sue here, but maybe `overlap_slices`...